crossdock/src/helpers.js

```diff
@@ -20,10 +20,188 @@
+            });
+        }
+    }
+
+    callDownstreamHTTP(downstream: Downstream, context: any): any {
+        return new RSVP.Promise((resolve, reject) => {
+
+            // $FlowIgnore - Honestly don't know why flow compalins about family.
+            dns.lookup(downstream.host, (err, address, family) => {
+                if (err) {
+                    console.log('dns_lookup_err', err);
+                    return;
+                }
+
+                let port = parseInt(downstream.port);
+                let downstreamUrl = `http:\/\/${address}:${port}/join_trace`;
+
+                let clientSpan = this._tracer.startSpan('client-span', { childOf: context.span.context() });
+                let headers = { 'Content-Type': 'application/json' };
+                this._tracer.inject(clientSpan.context(), opentracing.FORMAT_HTTP_HEADERS, headers);
+
+                request.post({
+                    'url': downstreamUrl,
+                    'forever': true,
+                    'headers': headers,
+                    'body': JSON.stringify({
+                        'serverRole': downstream.serverRole,
+                        'downstream': downstream.downstream
+                    })
+                }, (err, response) => {
+                    if (err) {
+                        console.log('error in downstream call:', err);
+                        reject(err);
+                        clientSpan.finish();
+                        return;
+                    }
+
+                    clientSpan.finish();
+                    let downstreamResponse = JSON.parse(response.body);
+                    resolve(downstreamResponse);
+                });
+
+            });
+        });
+    }
+
+    callDownstreamTChannel(downstream: Downstream, context: any): any {
+        return new RSVP.Promise((resolve, reject) => {
+            dns.lookup(downstream.host, (err, address, family) => {
+                if (err) {
+                    console.log('dns_lookup_err', err);
+                    return;
+                }
+
+                let port = parseInt(downstream.port);
+                let downstreamUrl = `http:\/\/${address}:${port}/join_trace`;
+
+                let clientSpan = this._tracer.startSpan('client-span', { childOf: context.span.context() });
+
+                // Save client span to tchannel headers
+                let headers = { 'Content-Type': 'application/json' };
+                TChannelBridge.saveTracerStateToCarrier(clientSpan, headers);
+                TChannelBridge.saveBaggageToCarrier(clientSpan, headers);
+
+                let request = this._thriftChannel.request({
+                    timeout: 2000,
+                    parent: { span: TChannelBridge.getTChannelParentSpan() },
+                    headers: {
+                        cn: 'tcollector-requestor'
+                    },
+                    trace: true,
```

Comment:
This forces trace, so that in tests we can look at tcap, and verify that a traceid of 0 is being sent.  (Not sure how to test that condition in a unit test, since I don't think I have access to the outgoing frame).